providers/implementations/ciphers/cipher_null.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include "prov/implementations.h"
+#include "prov/ciphercommon.h"
+#include "prov/providercommonerr.h"
+
+static OSSL_OP_cipher_newctx_fn null_newctx;
+static void *null_newctx(void *provctx)
+{
+    return OPENSSL_zalloc(1);
+}
+
+static OSSL_OP_cipher_freectx_fn null_freectx;
+static void null_freectx(void *vctx)
+{
+    OPENSSL_free(vctx);
+}
```

Comment:
Since the`null_newctx()` and `null_freectx()` calls are paired, and since the data pointed to by`vctx` is not used by the NULL cipher anyway, I don't see any need to allocate the data from the heap. All that is required is to return a valid pointer:


```C
static OSSL_OP_cipher_newctx_fn null_newctx;
static void *null_newctx(void *provctx)
{
    static int dummy = 0;
    return &dummy;
}

static OSSL_OP_cipher_freectx_fn null_freectx;
static void null_freectx(void *vctx)
{
    (void)vctx; /* unused */
}
```

Reply:
Good idea. I've done that.

Reply:
Fixup LGTM.